flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatMonitorImpl.java

```diff
@@ -70,6 +81,11 @@
                 "The heartbeat timeout interval has to be larger than 0.");
         this.heartbeatTimeoutIntervalMs = heartbeatTimeoutIntervalMs;
 
+        Preconditions.checkArgument(
+                failedRpcRequestsUntilUnreachable > 0 || failedRpcRequestsUntilUnreachable == -1,
+                "The number of failed heartbeat RPC requests has to be larger than 0 or -1 (deactivated).");
```

Comment:
Shouldn't we check this in the HeartbeatServices when we actually access the configuration?

Reply:
I think this makes sense, yes. I'll add it there as well.